scarlet/lite/component.py

```diff
@@ -0,0 +1,621 @@
+    _grad = np.einsum("i,i...", sed, _grad) * -morph
+    dY0 = ellipse.grad_y0(_grad, True)
+    dX0 = ellipse.grad_x0(_grad, True)
+    dSigmaY = ellipse.grad_major(_grad, True)
+    dSigmaX = ellipse.grad_minor(_grad, True)
+    dTheta = ellipse.grad_theta(_grad, True)
+    return np.array([dY0, dX0, dSigmaY, dSigmaX, dTheta], dtype=params.dtype)
+
+
+def circular_gaussian(center, frame, sigma):
+    """Model of a circularly symmetric Gaussian
+
+    Parameters
+    ----------
+    center: `numpy.ndarray`
+        The center of the Gaussian.
+    frame: `CartesianFrame`
+        The frame in which to generate the image of the circular Gaussian
+    sigma: `flaot`
+        The standard deviation.
+
+    Returns
+    -------
+    result: `numpy.ndarray`
+        The image of the circular Gaussian.
+    """
+    y0, x0 = center[:2]
+    two_sigma = 2 * sigma
+    r2 = ((frame.X - x0) / two_sigma) ** 2 + ((frame.Y - y0) / two_sigma) ** 2
+    return np.exp(-r2)
+
+
+def grad_circular_gaussian(input_grad, params, cls, morph, sed, frame, sigma):
+    """Gradient of the the component model wrt the Gaussian morphology parameters
+
+    Parameters
+    ----------
+    input_grad: `numpy.ndarray`
+        Gradient of the likelihood wrt the component model
+    params: `numpy.ndarray`
+        The parameters of the morphology.
+    cls: `LiteComponent`
+        The component of the model that contains the morphology.
+    morph: `numpy.ndarray`
+        The model of the morphology.
+    sed: `numpy.ndarray`
+        The model of the SED.
+    frame: `CartesianFrame`
+        The frame in which to generate the image of the circular Gaussian.
+    """
+    # Calculate the gradient of the likelihod
+    # wrt the Gaussian e^-r**2
+    _grad = np.zeros(cls.bbox.shape, dtype=morph.dtype)
+    _grad[cls.slices[1]] = input_grad[cls.slices[0]]
+    _grad = np.einsum("i,i...", sed, _grad) * -morph
+
+    y0, x0 = params[:2]
+    dY0 = np.sum(-2 * (frame.Y - y0) * _grad)
+    dX0 = np.sum(-2 * (frame.X - x0) * _grad)
+    return np.array([dY0, dX0], dtype=params.dtype)
+
+
+def integrated_gaussian(params, frame):
+    """Model of a circularly symmetric Gaussian integrated over pixels
+
+    This differs from `circularGaussian` because the gaussian function
+    is integrated over each pixel to replicated the pixelated image
+    version of a Gaussian function.
+
+    Parameters
+    ----------
+    params: `numpy.ndarray`
+        The center of the Gaussian.
+    frame: `CartesianFrame`
+        The frame in which to generate the image of the circular Gaussian
+
+    Returns
+    -------
+    result: `numpy.ndarray`
+        The image of the circular Gaussian.
+    """
+    # Unpack the parameters and define constants
+    y0, x0, sigma = params
+    r = np.sqrt((frame.X-x0)**2 + (frame.Y-y0)**2)
+    sqrt_c = 1 / np.sqrt(2) / sigma
+    # Integrate from half a pixel left and right
+    lhs = erf((r - 0.5) * sqrt_c)
+    rhs = erf((r + 0.5) * sqrt_c)
```

Comment:
I don't think so. ERF integrates from zero to `A`, so you need two in order to integrate from `A` to `B` (0 to `B` - 0 to `A`). Even if numpy/scipy had an integral from `A` to `B` it would functionally be the same thing.